projects/ngx-prx-styleguide/src/lib/hal/remote/halremote.spec.ts

```diff
@@ -207,19 +201,29 @@ describe('HalRemote', () => {
   });
 
   describe('retries', () => {
-    it('retries 401s after getting a new token', () => {
-      jest.spyOn(fakeAuth, 'refreshToken');
-      remote.get(link).subscribe();
-
-      const httpCount = 0;
-      const req = mockHttp.expectOne(request => request.headers.get('Authorization') === 'Bearer thetoken');
-      expect(req.request.method).toBe('GET');
-      req.flush({ data: 'Unauthorized' }, { status: 401, statusText: 'Unauthorized' });
-
-      expect(fakeAuth.refreshToken).toHaveBeenCalledTimes(1);
-      token.subscribe(tkn => {
-        expect(tkn).toEqual('nexttoken');
-      });
+    it('retries 401s exactly once after getting a new token', () => {
+      let result: any;
+      remote.get(link).subscribe(r => (result = r));
```

Comment:
The real trick was ... this `subscribe` only gets called once, even if we were erroneously replaying the HTTP request twice.

That's because `get()` was already guarded by a `first()`.  So we were sort of silently replaying HTTP requests without the caller app being aware of it.